File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/views/setup/AvatarCategoryView.java
package com.habitrpg.android.habitica.ui.views.setup;


import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.support.annotation.Nullable;
import android.support.v4.content.ContextCompat;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.habitrpg.android.habitica.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public class AvatarCategoryView extends LinearLayout {

    private final Drawable icon;
    @BindView(R.id.text_view)
    TextView textView;

    public AvatarCategoryView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        inflate(context, R.layout.avatar_category, this);

        ButterKnife.bind(this);

        TypedArray a = context.getTheme().obtainStyledAttributes(
                attrs,
                R.styleable.AvatarCategoryView,
                0, 0);

        textView.setText(a.getText(R.styleable.AvatarCategoryView_categoryTitle));

        icon = a.getDrawable(R.styleable.AvatarCategoryView_iconDrawable);
        if (icon != null) {
            textView.setCompoundDrawablesWithIntrinsicBounds(null, icon, null, null);
        }
        setActive(false);
    }

    public void setActive(boolean active) {
        int color;
        if (active) {
            color = ContextCompat.getColor(getContext(), R.color.white);
        } else {
            color = ContextCompat.getColor(getContext(), R.color.white_50_alpha);
        }
        textView.setTextColor(color);
        if (icon != null) {
            icon.setColorFilter(color, PorterDuff.Mode.MULTIPLY);
            textView.setCompoundDrawablesWithIntrinsicBounds(null, icon, null, null);
        }
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/views/setup/AvatarCategoryView.kt
package com.habitrpg.android.habitica.ui.views.setup

import android.content.Context
import android.graphics.PorterDuff
import android.graphics.drawable.Drawable
import android.util.AttributeSet
import android.view.View
import android.widget.LinearLayout
import android.widget.TextView
import androidx.core.content.ContextCompat
import com.habitrpg.android.habitica.R
import com.habitrpg.common.habitica.extensions.setTintWith

class AvatarCategoryView(context: Context, attrs: AttributeSet?) : LinearLayout(context, attrs) {

    private val icon: Drawable?
    private val textView: TextView

    init {
        View.inflate(context, R.layout.avatar_category, this)

        textView = findViewById(R.id.text_view)
        val a = context.theme.obtainStyledAttributes(
            attrs,
            R.styleable.AvatarCategoryView,
            0,
            0
        )

        textView.text = a.getText(R.styleable.AvatarCategoryView_categoryTitle)

        icon = a.getDrawable(R.styleable.AvatarCategoryView_iconDrawable)
        if (icon != null) {
            textView.setCompoundDrawablesWithIntrinsicBounds(null, icon, null, null)
        }
        setActive(false)
    }

    fun setActive(active: Boolean) {
        val color: Int = if (active) {
            ContextCompat.getColor(context, R.color.white)
        } else {
            ContextCompat.getColor(context, R.color.white_50_alpha)
        }
        textView.setTextColor(color)
        if (icon != null) {
            icon.setTintWith(color, PorterDuff.Mode.MULTIPLY)
            textView.setCompoundDrawablesWithIntrinsicBounds(null, icon, null, null)
        }
    }
}
